src/EventSubscriber/TideSiteGetRouteSubscriber.php

```diff
@@ -198,6 +206,46 @@ public function onApiGetRouteAddSiteFilter(GetRouteEvent $event) {
     }
   }
 
+  /**
+   * Process the redirect from the GetRoute event.
+   *
+   * @param \Drupal\tide_api\Event\GetRouteEvent $event
+   *   The event object.
+   */
+  protected function processGetRouteRedirect(GetRouteEvent $event) {
+    $response = $event->getJsonResponse();
+    // Only process the redirect to internal paths.
+    if ($response['data']['attributes']['redirect_type'] == 'internal') {
+      $redirect_url = $response['data']['attributes']['redirect_url'];
+      // This destination path does not have the site prefix, bail out early.
+      if (!$this->siteHelper->hasSitePrefix($redirect_url)) {
+        return;
+      }
+
+      // Retrieve the site prefix from the redirect.
+      if (preg_match("/^\/site\-(\d+)\//", $redirect_url, $matches) && !empty($matches[1])) {
+        $destination_site = $this->siteHelper->getSiteById($matches[1]);
+        if ($destination_site) {
+          $prefix = $this->siteHelper->getSitePathPrefix($destination_site);
+          $redirect_url = str_replace($prefix, '', $redirect_url);
+          $current_site_id = $event->getRequest()->query->get('site');
+          // Remove the site prefix if the redirect belongs to the same site.
+          if ($current_site_id == $destination_site->id()) {
+            $response['data']['attributes']['redirect_url'] = $redirect_url;
+          }
+          // Otherwise treat it as an external path, and return the full URL
+          // of the destination site.
+          else {
+            $response['data']['attributes']['redirect_type'] = 'external';
+            $response['data']['attributes']['redirect_url'] = $this->siteHelper->getSiteBaseUrl($destination_site) . $redirect_url;
+          }
+          $event->setJsonResponse($response);
+          $event->getCacheableMetadata()->addCacheableDependency($destination_site);
```

Comment:
Why don't we need to do the cache_tags thing here?

Reply:
`$destination_site` is a cachable object, hence it can be added directly to the cachablemetadata.